src/ode_solver/explicit_ode_solver.cpp

```diff
@@ -110,14 +110,14 @@ template <int dim, typename real, int n_rk_stages, typename MeshType>
 void RungeKuttaODESolver<dim,real,n_rk_stages,MeshType>::allocate_ode_system ()
 {
     this->pcout << "Allocating ODE system and evaluating inverse mass matrix..." << std::endl;
-    this->solution_update.reinit(this->dg->right_hand_side);
+    this->solution_update.reinit(this->dg->locally_owned_dofs, this->dg->ghost_dofs, this->mpi_communicator);
```

Comment:
Why are those changes related to large memory allocation?

Reply:
https://www.dealii.org/current/doxygen/deal.II/group__Vectors.html#ga3be6c4ce529bb9b6c13eb831d0a86f55 
When initialized with the global size rather than the partitioning info, then it doesn't do any memory partitioning.

Reply:
`this->dg->right_hand_side` is not a global size, it is a vector.

It uses the `reinit` function akin to the copy constructor
https://www.dealii.org/current/doxygen/deal.II/group__Vectors.html#ga8a79ca37d7f7a8cfd04f5dd61d350ae5
such that the `solution_update` will have the same parallel partitioning as `right_hand_size`

Can we revert this change?

Reply:
Ok yes let's revert it. I'm no longer able to replicate the memory jump. The memory fix must've been from the commit before and I didn't see it from the build. Thanks!

Reply:
fixed